java/client/src/main/java/glide/api/RedisClient.java

```diff
@@ -78,4 +87,118 @@ public CompletableFuture<Object> customCommand(String[] args) {
+        return commandManager.submitNewCommand(command, BaseClient::handleMapResponse);
+    }
+
+    /**
+     * Get information and statistics about the Redis server.
+     *
+     * @see <a href="https://redis.io/commands/info/">redis.io</a> for details.
+     * @param options - A list of InfoSection values specifying which sections of information to
+     *     retrieve. When no parameter is provided, the default option is assumed.
+     * @return CompletableFuture with the response
+     */
+    @Override
+    public CompletableFuture<Map> info(InfoOptions options) {
+        Command command =
+                Command.builder()
+                        .requestType(Command.RequestType.INFO)
+                        .arguments(options.toInfoOptions())
+                        .build();
+        return commandManager.submitNewCommand(command, BaseClient::handleMapResponse);
+    }
+
+    /**
+     * Get the value associated with the given key, or null if no such value exists.
+     *
+     * @see <a href="https://redis.io/commands/get/">redis.io</a> for details.
+     * @param key - The key to retrieve from the database.
+     * @return If `key` exists, returns the value of `key` as a string. Otherwise, return null
+     */
+    @Override
+    public CompletableFuture<String> get(String key) {
+        Command command =
+                Command.builder()
+                        .requestType(Command.RequestType.GET_STRING)
+                        .arguments(new String[] {key})
+                        .build();
+        return commandManager.submitNewCommand(command, BaseClient::handleStringResponse);
+    }
+
+    /**
+     * Set the given key with the given value.
+     *
+     * @see <a href="https://redis.io/commands/set/">redis.io</a> for details.
+     * @param key - The key to store.
+     * @param value - The value to store with the given key.
+     * @return null
+     */
+    @Override
+    public CompletableFuture<Void> set(String key, String value) {
+        Command command =
+                Command.builder()
+                        .requestType(Command.RequestType.SET_STRING)
+                        .arguments(new String[] {key, value})
+                        .build();
+        return commandManager.submitNewCommand(command, BaseClient::handleVoidResponse);
+    }
+
+    /**
+     * Set the given key with the given value. Return value is dependent on the passed options.
+     *
+     * @see <a href="https://redis.io/commands/set/">redis.io</a> for details.
+     * @param key - The key to store.
+     * @param value - The value to store with the given key.
+     * @param options - The Set options
+     * @return string or null If value isn't set because of `onlyIfExists` or `onlyIfDoesNotExist`
+     *     conditions, return null. If `returnOldValue` is set, return the old value as a string.
+     */
+    @Override
+    public CompletableFuture<String> set(String key, String value, SetOptions options) {
```

Comment:
```suggestion
    public CompletableFuture<String> set(String key, String value, @NonNull SetOptions options) {
```

You will have NPE with and without `@NonNull` 🤔 But maybe worth adding `@NonNull` everywhere for all commands?